### PR TITLE
Refactor of HomePage component and directory structure

### LINKED ISSUE #1087: Use simple and clear React.js components
This issue is part of a frontend code review, see #1083.

Most components in the front ende codebase don’t do one thing, but several at once. This is not a good technique to follow. Because of this any future changes or improvements are extremely hard and a chance to make a mistake is much higher.

For example, the [CommitsTable](https://github.com/versionpress/versionpress/blob/master/frontend/src/Commits/CommitsTable.react.tsx) component is a very large one. The component can be very hard to follow and its formating style can make the component unreadable. 
The component can be split into smaller ones to improve its readability and clarity. The component even uses [CommitsTableNote](https://github.com/versionpress/versionpress/blob/master/frontend/src/Commits/CommitsTableNote.react.tsx) component, which is much easier to follow and does just a one thing.

### PR DESCRIPTION
**Pull request by Ishirak**
Resolves #1087.

Last piece to close #1087 – major refactor of front end codebase will be over.

Reviewers:

- [x] @JanVoracek 
- [x] @vasek17

### CODE REVIEW

File: frontend/src/components/home/HomePage.tsx [hunk 1]
```diff
@@ -136,7 +136,6 @@ export default class HomePage extends React.Component<HomePageProps, HomePageSta
             isLoading: false,
             displayUpdateNotice: false,
           });
-          this.checkUpdate();
```
Comment (vasek17) [inline frontend/src/components/home/HomePage.tsx]: @Ishirak: Why is this line removed? If the checkUpdate is not called after fetching commits there could be short (<10s) inconsistency between the commits table and the CommitPanel.

Comment (Ishirak) [inline frontend/src/components/home/HomePage.tsx]: You are right, I thought that in this case `setState()` would behave as an asynchronous method, as usual, but it doesn't, as I just checked. I will revert this change in the next commit.
